src/muiComponents/Heading/Heading.tsx

```diff
@@ -0,0 +1,19 @@
+import React, { forwardRef } from 'react';
+import { default as MaterialUITypography, TypographyProps } from '@material-ui/core/Typography';
+
+type HeadingType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+type HeadingRef = Extract<keyof HTMLElementTagNameMap, HeadingType>;
```

Comment:
`HeadingRef` has the same type as `HeadingType`

Reply:
Hm...thank you for your feedback, but unfortunately that is not exactly want I want..please see:

![image](https://user-images.githubusercontent.com/29228205/66684916-7041be00-ec7b-11e9-86ad-7614d2ae4c15.png)

Reply:
You are seeing the definition of `HeadingRef` but this will in fact evaluate to `HTMLHeadingElement` ;)

```typescript
type HeadingType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
type HeadingRef<T extends keyof HTMLElementTagNameMap = HeadingType> = HTMLElementTagNameMap[T];
type HeadingElements = HeadingRef; // -> its evaluated type is HTMLHeadingElement
```
Or just hover over `forwardRef` :)